src/app/shared/object-collection/shared/listable-object/listable-object-component-loader.component.ts

```diff
@@ -166,27 +165,32 @@ export class ListableObjectComponentLoaderComponent implements OnInit, OnChanges
 
     const component = this.getComponent(object.getRenderTypes(), this.viewMode, this.context);
 
-    const componentFactory = this.componentFactoryResolver.resolveComponentFactory(component);
-
     const viewContainerRef = this.listableObjectDirective.viewContainerRef;
     viewContainerRef.clear();
 
     this.compRef = viewContainerRef.createComponent(
-      componentFactory,
-      0,
-      undefined,
-      [
-        [this.badges.nativeElement],
-      ]);
+      component, {
+        index: 0,
+        injector: undefined,
+        projectableNodes: [
+          [this.badges.nativeElement],
+        ]
+      }
+    );
 
     this.connectInputsAndOutputs();
 
     if ((this.compRef.instance as any).reloadedObject) {
-      (this.compRef.instance as any).reloadedObject.pipe(take(1)).subscribe((reloadedObject: DSpaceObject) => {
+      (this.compRef.instance as any).reloadedObject.pipe(
+        // Add delay before emitting event to allow the new object is elaborated on REST side
+        debounceTime((100)),
+        take(1)
+      ).subscribe((reloadedObject: DSpaceObject) => {
         if (reloadedObject) {
           this.compRef.destroy();
           this.object = reloadedObject;
           this.instantiateComponent(reloadedObject);
+          this.cdr.detectChanges();
```

Comment:
What changed to make this call necessary now?
Please check if you can remove it again if you can get the caching issue fixed.

Reply:
this is neccessary otherwise sometimes new element does not appear immediatly, but after moving in the page. Anyway with or without the problem with cache persists

Reply:
Have you checked whether this is still the case after the refactor yesterday?
When I tested with `debounceTime` in the facet component locally, it seemed that this call was no longer required here.